src/agenda/reservation-list/reservation.tsx

```diff
@@ -75,8 +76,8 @@ class Reservation extends Component<ReservationProps> {
   }
 
   renderDate(date?: XDate, item?: DayReservations) {
-    if (isFunction(this.props.renderDay) && date) {
-      return this.props.renderDay(date, item);
+    if (isFunction(this.props.renderDay)) {
+      return this.props.renderDay(date ? xdateToData(date) : undefined, item);
```

Comment:
You're changing the return value, it should be XDate, not DateData object

Reply:
You'll also need to change type on line 23 to `renderDay?: (**date?**: XDate, item?: DayReservations) => React.Component;`

Reply:
Actually I should only pass date which is `XDate`, same as the type required in `renderDay`.